### PR TITLE
Fstring gem

### PR DESCRIPTION
**Pull request by schneems**
Locally I'm seeing this change makes the homepage faster:

```
❤️ ❤️ ❤️  (Statistically Significant) ❤️ ❤️ ❤️

[59cb042] "Fstring gem" - (4.0876645 seconds)
  FASTER 🚀🚀🚀 by:
    1.0100x [older/newer]
    0.9918% [(older - newer) / older * 100]
[d9ed02a] "Merge pull request #1226 from codetriage/schneem/remove-fragment-caching" - (4.1286115 seconds)

Iterations per sample: 200
Samples: 200

Test type: Kolmogorov Smirnov
Confidence level: 99.0 %
Is significant? (max > critical): true
D critical: 0.15174271293851463
D max: 0.175

Output: tmp/compare_branches/2020-01-01-13-10-1577905803-361481000/results.txt
```

### CODE REVIEW

**Comment by schneems** [comment]
I think something is fundamentally wrong with the way that we're doing comparisons. I ran the same command 4 times in a row on my computer after closing all apps and this is what i'm seeing:


```
👎👎👎(NOT Statistically Significant) 👎👎👎

[6aab468] "Fstring gem" - (3.35021421 seconds)
  SLOWER by:
    0.9979x [older/newer]
    -0.2081% [(older - newer) / older * 100]
[c321bd2] "Merge pull request #1188 from codetriage/schneems/upsert-doc-methods" - (3.34325846 seconds)

Iterations per sample: 200
Samples: 200

Test type: Kolmogorov Smirnov
Is significant? (max > critical): false
D critical: 0.12238734153404084
D max: 0.07500000000000001
```



```          
👎👎👎(NOT Statistically Significant) 👎👎👎

[6aab468] "Fstring gem" - (3.38259895 seconds)
  SLOWER by:
    0.9884x [older/newer]
    -1.1725% [(older - newer) / older * 100]
[c321bd2] "Merge pull request #1188 from codetriage/schneems/upsert-doc-methods" - (3.34339648 seconds)

Iterations per sample: 200
Samples: 200

Test type: Kolmogorov Smirnov
Is significant? (max > critical): false
D critical: 0.12238734153404084
D max: 0.11499999999999999
```



```
❤️ ❤️ ❤️  (Statistically Significant) ❤️ ❤️ ❤️

[6aab468] "Fstring gem" - (4.300999547945206 seconds)
  SLOWER by:
    0.8379x [older/newer]
    -19.3480% [(older - newer) / older * 100]
[c321bd2] "Merge pull request #1188 from codetriage/schneems/upsert-doc-methods" - (3.6037456164383563 seconds)

Iterations per sample: 200
Samples: 73

Test type: Kolmogorov Smirnov
Is significant? (max > critical): true
D critical: 0.2025769690879227
D max: 0.31506849315068486
```




```
❤️ ❤️ ❤️  (Statistically Significant) ❤️ ❤️ ❤️

[6aab468] "Fstring gem" - (3.3661813218390804 seconds)
  SLOWER by:
    0.9751x [older/newer]
    -2.5495% [(older - newer) / older * 100]
[c321bd2] "Merge pull request #1188 from codetriage/schneems/upsert-doc-methods" - (3.282494655172414 seconds)

Iterations per sample: 200
Samples: 87

Test type: Kolmogorov Smirnov
Is significant? (max > critical): true
D critical: 0.18556320835155882
D max: 0.37931034482758624
```

> Here is a zip of 4 different test runs raw data: https://www.dropbox.com/s/idak3kkqxz0tm2h/4_runs_derailed_results.zip?dl=0

**Comment by schneems** [comment]
I switched the comparison from using average to median I then re-ran the test 10 times and it's reported here https://github.com/schneems/derailed_benchmarks/issues/162

TLDR is that it looks like it might be statistically faster, but it's borderline. It's probably good enough to be merged in, but it's not a huge slam dunk like the 17% indicated previously. Based on this experience I'm going to make changes to derailed to try to increase confidence.
